src/arrayMethodSort.js

```diff
@@ -4,8 +4,35 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compareFn =
+      compareFunction ||
+      function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+
+    const len = this.length;
+
+    for (let i = 0; i < len; i++) {
+      for (let j = 0; j < len - 1 - i; j++) {
```

Comment:
The inner loop condition can be optimized. Since the largest element is guaranteed to be in its correct position after each pass, you can reduce the number of comparisons by subtracting 'i' from the length in the condition: 'len - 1 - i'.